application/controllers/LoginManager.php

```diff
@@ -18,7 +18,92 @@ function force_ssl() {
 	    // }
 	}
 
+	function handleLocalLoginForVueUI() {
+		$requestMethod = $this->input->server('REQUEST_METHOD');
+		if ($requestMethod == 'GET') {
+			return $this->template->publish('vueTemplate');
+		}
+
+		// respond with json
+		header('Content-Type: application/json');
+
+		// if this isn't a POST request, return an error
+		if ($requestMethod != 'POST') {
+			http_response_code(405);
+			echo json_encode([
+				'status' => 'error',
+				'message' => 'invalid request method'
+			]);
+			return;
+		}
+
+		// Get the data from the request body
+		$cleanJSON = $this->security->xss_clean($this->input->raw_input_stream);
```

Comment:
I'd rather this be a normal post (even if it's Ajax) than a JSON bundle. We're not doing bare JSON payloads anywhere else in the codebase, so this becomes an outlier. I don't want to trust a 10 year old xss_clean method from CI to actually be robust either. Would much rather stick with traditional post fields and pull them out in the standard way - it doesn't seem like the JSON is providing any benefit here over a form?